Deeper gts support broken in default install

Reproduction repo: https://github.com/ef4/glint-eslint-bug
Reproduction commit: https://github.com/ef4/glint-eslint-bug/commit/7036d8277be29b2ae8499b9cbefd6da792040f87
Behavior: `pnpm lint:js` fails with

```
/glint-eslint-bug/app/two.ts
  3:1  error  Unsafe call of a type that could not be resolved  @typescript-eslint/no-unsafe-call
```

This case only fails when we're importing a type from a GTS file *through* a TS file. If we directly import the type from a GTS file it works as expected.

My reproduction used a newly-generate app on ember 7. which uses eslint-plugin-ember "^12.7.5". But I also attempted upgrading eslint-plugin-ember to `^13.0.0` and the bug remains.

--- a/package.json
+++ b/package.json
@@ -67,7 +67,7 @@
     "axobject-query": "^4.1.0",
     "css-tree": "^3.0.1",
     "editorconfig": "^3.0.2",
-    "ember-eslint-parser": "^0.12.0",
+    "ember-eslint-parser": "^0.12.1",
     "ember-rfc176-data": "^0.3.18",
     "eslint-utils": "^3.0.0",
     "estraverse": "^5.3.0",

--- a/tests/lib/rules-preprocessor/gjs-gts-parser-test.js
+++ b/tests/lib/rules-preprocessor/gjs-gts-parser-test.js
@@ -184,8 +184,8 @@ const invalid = [
     }`,
     errors: [
       {
-        message: `Parsing error: × Expected ';', got 'MyComponent'
-   ╭─[1:1]
+        message: `Parsing error: × Expected ';', got 'ident'
+   ╭─[2:1]
  1 │ import Component from '@glimmer/component';
  2 │     export default classsss MyComponent extends Component {
    ·                             ───────────

